stdlib/StdlibModules

```diff
@@ -34,17 +34,72 @@
 # Basenames of the source files for the standard library (i.e. unprefixed and
 # with lowercase first letters). These must be listed in dependency order.
 STDLIB_MODULE_BASENAMES = \
-  camlinternalFormatBasics stdlib either \
-  sys obj atomic camlinternalLazy lazy \
-  seq option result bool char uchar \
-  list int bytes string unit marshal array float int32 int64 nativeint \
-  lexing parsing set map stack queue buffer \
-  mutex condition semaphore domain \
-  camlinternalFormat printf arg printexc \
-  fun gc digest bigarray random hashtbl weak \
-  format scanf callback camlinternalOO oo camlinternalMod ephemeron \
-  filename complex arrayLabels listLabels bytesLabels stringLabels moreLabels \
-  stdLabels in_channel out_channel effect
+  type \
```

Comment:
Is it ok to put type before `stdlib` ?

Reply:
scratch that :-) rather is ok to have stdlib that further up.

Reply:
Well, perhaps it works because `Type` currently does not depend on anything at all, but the convention is that we put all the "wrapped" modules after `stdlib` so I moved it.

> scratch that :-) rather is it ok to have stdlib that further up.

This is always a bit confusing, but I think that the `-no-alias-deps` stuff means that `stdlib` does not actually depend on any other module (while in turn every other module depends on it to resolve the "wrapped" names).